src/dbnode/persist/fs/seek.go

```diff
@@ -224,15 +225,18 @@ func (s *seeker) Open(
 	}
 	s.start = xtime.UnixNano(info.BlockStart)
 	s.blockSize = time.Duration(info.BlockSize)
+	s.versionChecker = schema.NewVersionChecker(int(info.MajorVersion), int(info.MinorVersion))
 
-	err = s.validateIndexFileDigest(
-		indexFdWithDigest, expectedDigests.indexDigest)
-	if err != nil {
-		s.Close()
-		return fmt.Errorf(
-			"index file digest for file: %s does not match the expected digest: %c",
-			filesetPathFromTimeLegacy(shardDir, blockStart, indexFileSuffix), err,
-		)
+	if !s.versionChecker.IndexEntryValidationEnabled() {
```

Comment:
TODO on me to update `seek_test.go` to check for this, but wanted to get eyeballs on the PR first.